### PR TITLE
Adds gamer flare to the game.

### PR DESCRIPTION
**Pull request by moonheart08**
Introduces the RgbLightController component, an adjustable-rate oscillator that cycles through various hues.

**Screenshots**
Screenshot from 2021-10-23 19-13-46

### CODE REVIEW

File: Content.Shared/Light/Component/SharedRgbLightControllerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Shared.Light.Component
+{
+    [NetworkedComponent]
+    [RegisterComponent]
+    public class SharedRgbLightControllerComponent : Robust.Shared.GameObjects.Component
```
Comment (metalgearsloth) [inline Content.Shared/Light/Component/SharedRgbLightControllerComponent.cs:10]: yuck.

Also don't need share prefix if it's not being inherited.

File: Content.Shared/Light/SharedRgbLightControllerSystem.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            foreach (var (rgb, light) in EntityManager.EntityQuery<SharedRgbLightControllerComponent, SharedPointLightComponent>())
+            {
+                light.Color = GetCurrentRgbColor(_gameTiming, TimeSpan.FromSeconds(rgb.CreationTick.Value * _gameTiming.TickPeriod.TotalSeconds), rgb);
+            }
+        }
```
Comment (metalgearsloth) [inline Content.Shared/Light/SharedRgbLightControllerSystem.cs:22]: Still server-states going out?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Client/Light/RgbLightControllerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            {
+                light.Color = GetCurrentRgbColor(_gameTiming, TimeSpan.FromSeconds(rgb.CreationTick.Value * _gameTiming.TickPeriod.TotalSeconds), rgb);
+                sprite.Color = GetCurrentRgbColor(_gameTiming, TimeSpan.FromSeconds(rgb.CreationTick.Value * _gameTiming.TickPeriod.TotalSeconds), rgb);
+            }
+        }
+
+        public static Color GetCurrentRgbColor(IGameTiming gameTiming, TimeSpan offset, RgbLightControllerComponent rgb)
+        {
+            return Color.FromHsv(new Vector4(
+                (float) (((gameTiming.CurTime.TotalSeconds - offset.TotalSeconds) / rgb.CycleRate + Math.Abs(rgb.Owner.Uid.GetHashCode() * 0.1)) % 1),
```
Comment (metalgearsloth) [inline Content.Client/Light/RgbLightControllerSystem.cs:30]: Should pass in currenttime as CurTime is a property that needs to be re-calculated every time (and it does show up in the profiler from time to time).

**Review by metalgearsloth — APPROVED** [review]

**Review by PaulRitter — APPROVED** [review]

**Review by PaulRitter — CHANGES_REQUESTED** [review]
change what sloth noted & this should be good to merge

**Pull request merged**